packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -80,6 +83,21 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
                   action: SmoothActionButton(
                     text: appLocalizations.finish,
                     onPressed: () async {
+                      final LocalDatabase localDatabase =
+                          context.read<LocalDatabase>();
+                      final DaoProduct daoProduct = DaoProduct(localDatabase);
+                      final Product? product =
+                          await daoProduct.get(widget.barcode);
+                      if (product == null) {
+                        final Product product = Product(
+                          barcode: widget.barcode,
+                        );
+                        daoProduct.put(product);
+                        localDatabase.notifyListeners();
```

Comment:
Shouldn't this be outside of the if and not overwrite the old product

Also what are the changes in this file for are the related to the background upload

Reply:
see, what I did is say I am still doing the work and say I uploaded an image then before I finish the button the product is already now in the local DB as it finished quickly. 
So I check if it's already saved,if it is then i don't do anything , 
if the task is not completed so I just show a placeholder product there for now.
I hope it clarified it a bit.